Hazardous Gases (Cl and HCl)

Thera-Pissed opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds chlorine gas and hydrogen chloride gas
These gases hurt to be exposed to in high concentration if not in EVA
Plasma, Trit, Cl, and HCl now damage eyes if not wearing goggles (welding, heat, or cold goggles currently)
Chlorine reacts with hydrogen to make HCl and makes heat while doing so. HCl burns into H2O and Cl.
![image](https://github.com/shiptest-ss13/Shiptest/assets/90987989/e0263c74-0958-475f-b956-4024b32ed237)
![image](https://github.com/user-attachments/assets/ac384206-c567-4baa-9fbc-42ec56a5e02e)

Scrubbers are now set to filter most gases at roundstart.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Hazardous planetary gases! (not actually implemented in this PR, but opens up the possibility)
More gases that are actually relevant to Normal shiptest rounds is good, since there's few station gases that can be reasonably seen or made (freon has been made like once. stimulum and nitryl are dumb. likely could be removed)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
add: Chlorine and Hydrogen Chloride gases! Wear your goggles!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Thera-Pissed committed:
first draft

Thera-Pissed committed:
fixes linter checks and tweaks thrust

Thera-Pissed committed:
linters take 2

Thera-Pissed committed:
adds overpressure damage, hydrogen gas miner, and a temporary fuel test canister

Thera-Pissed committed:
adds repair functionality, examine text coming soon

Thera-Pissed committed:
repair 2, examine steps, stockparts/board

examine tells next repair step

Thera-Pissed committed:
fixes dumb error, more stockparts stuff

Thera-Pissed committed:
adds combustion thrusters to shuttle research

Thera-Pissed committed:
thanks bog

Thera-Pissed committed:
ice ore spawns now

Thera-Pissed committed:
adds unique sprites for ice ore scanning and items

these are mid and will need to be replaced probably

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'Thruster' of https://github.com/Thera-Pissed/Shiptest into Thruster

Thera-Pissed committed:
I lied. removes some testing stuff and an icon fix

Thera-Pissed committed:
instructions, open turf tanks, canister, safe limit on examine

Thera-Pissed committed:
adds electrolyzers to outpost for TMing

Thera-Pissed committed:
lights + signs

Thera-Pissed committed:
integration tests fix?

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
switch instead of if chain

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
linters fix I hope

Thera-Pissed committed:
adds two new canister sprites for fuel & hydrogen

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
kill indie map in an attempt to put in the new one

Thera-Pissed committed:
50/50 shot this fixes it or breaks my maps

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
fixes map merge stuff

Thera-Pissed committed:
Merge branch 'Thruster' of https://github.com/Thera-Pissed/Shiptest into Thruster

Thera-Pissed committed:
fixes random doubled tiles

hopefully there's none that dont cause linter fails!

Thera-Pissed committed:
fixes evil doubled tiles for real

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

forgot to change that after I got it working

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/modules/atmospherics/machinery/portable/canister.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/modules/mining/ores_coins.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/modules/research/designs/machine_designs.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/modules/research/designs/machine_designs.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/modules/research/designs/machine_designs.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
makes MOLS_PER_ICE do anything

Thera-Pissed committed:
Merge branch 'Thruster' of https://github.com/Thera-Pissed/Shiptest into Thruster

Thera-Pissed committed:
actually fixes MOLS_PER_ICE

Thera-Pissed committed:
Update code/game/machinery/shuttle/shuttle_engine_types.dm

Co-authored-by: JimKil3 <47290811+JimKil3@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
addresses review stuff!

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Update code/game/machinery/outpost_electrolyzer.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Update code/game/machinery/shuttle/shuttle_engine_types.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
most of the review

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
resolves map conflict?

Thera-Pissed committed:
Merge branch 'master' into Thruster

Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
attempts to revert indie_space to fix problems. might not

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
fixes indie_space changes pt 6

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
reverts map changes. again. hopefully

Thera-Pissed committed:
reverts but from a modern commit? maybe?

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
scrip replaces direct production

Thera-Pissed committed:
slightly nicer placeholder sprites

Thera-Pissed committed:
hydrogen exchange gets worse the more its used

Thera-Pissed committed:
better placeholder pump sprite

Thera-Pissed committed:
noises and stuff

Thera-Pissed committed:
pump is on click instead of ctrl click

Thera-Pissed committed:
outpost tax rate to encourage hydrogen mining

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
reverts minerals.dm for now

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
addresses review + h exchange sprite

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into Thruster

Thera-Pissed committed:
resolves conflict hopefully

Thera-Pissed committed:
github eludes me still

Thera-Pissed committed:
fixes conflict

Thera-Pissed committed:
re-adds placeholder pump icon

Thera-Pissed committed:
im skill

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
I sure hope it works this time

Thera-Pissed committed:
makes ice ore show up again (only on ice planets, duh)

Thera-Pissed committed:
electrolyzer eats ice again, oopsies

Thera-Pissed committed:
stuff I missed, for review

Thera-Pissed committed:
actually addresses review of outpost_electrolyzer. oops

Thera-Pissed committed:
adds combustion thruster boards for 2000 credits

Thera-Pissed committed:
adds combustion thrusters to the dwayne, skipper, and crying sun for TM.

Thera-Pissed committed:
minor ship tweaks

Thera-Pissed committed:
reverts computer dmi to fix conflict

Thera-Pissed committed:
readds new screen

Thera-Pissed committed:
I hate code

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
if I just do the exact same thing again it'll work right.

Thera-Pissed committed:
adds combustion thrusters to inkwell and komodo

Thera-Pissed committed:
test your map changes. duh.

Thera-Pissed committed:
Merge branch 'master' into Thruster

Thera-Pissed committed:
implements PR #2977 heater layers

Thera-Pissed committed:
how'd that get there

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into Thruster

Thera-Pissed committed:
Merge commit '493ea5f' into tmp

Thera-Pissed committed:
Merge commit '493ea5f' into tmp

Thera-Pissed committed:
Merge branch 'tmp' into Thruster

Thera-Pissed committed:
what do you mean I should just ask someone who knows what they're doing. Im proficient

Thera-Pissed committed:
work

Thera-Pissed committed:
this should be fixed now

Thera-Pissed committed:
this is horrid

Thera-Pissed committed:
why couldn't it just let me right click revert, huh

goober3 committed:
outpost maps

goober3 committed:
Merge branch 'Thruster' of https://github.com/Thera-Pissed/Shiptest into pr/2431

Thera-Pissed committed:
fixes double pipe, pump broke still

Thera-Pissed committed:
bug 'fixed'

Thera-Pissed committed:
dumb stupid no-fun linters

Thera-Pissed committed:
another double pipe

Thera-Pissed committed:
wrong unloader

Thera-Pissed committed:
Update code/game/machinery/shuttle/shuttle_heater.dm

Co-authored-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
wrong path

Thera-Pissed committed:
fixes reconstructing heater bug for plasma & combustion and all thruster machines default to east on construction

Thera-Pissed committed:
probably should unregister the old signal huh

Thera-Pissed committed:
chlorine and hcl take 1

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into chlorine-hydrogen-chloride-time

Thera-Pissed committed:
dumb

Thera-Pissed committed:
chlorine is an oxidizer for thrusters

Thera-Pissed committed:
Merge branch 'master' into chlorine-hydrogen-chloride-time

Thera-Pissed commented:
woohoo 164 commits because I started before hydrogen got in!

rye-rice commented on code/modules/atmospherics/machinery/portable/canister.dm:
```diff
@@ -54,7 +54,9 @@
+		"fuel mix" = /obj/machinery/portable_atmospherics/canister/fuel,
+		"chlorine" = /obj/machinery/portable_atmospherics/canister/chlorine,
+		"hydrogen chloride" =/obj/machinery/portable_atmospherics/canister/hydrogen_chloride
 	)
```
o,
```suggestion
		"fuel mix" = /obj/machinery/portable_atmospherics/canister/fuel,
		"chlorine" = /obj/machinery/portable_atmospherics/canister/chlorine,
		"hydrogen chloride" =/obj/machinery/portable_atmospherics/canister/hydrogen_chloride,
	)
```

rye-rice reviewed (commented):
Push a new commit, seems checks failed for some unrelated reason

rye-rice commented:
oh wow that fixed the checks

Thera-Pissed committed:
Update code/modules/atmospherics/machinery/portable/canister.dm

Co-authored-by: rye-rice <58402542+rye-rice@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

SomeguyManperson commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -61,6 +61,21 @@
+		var/obj/item/clothing/CH = head
+		if (CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
+			return TRUE
+	return FALSE
```
```suggestion
	var/obj/item/clothing/CS = wear_suit
	var/obj/item/clothing/CH = head
	if(istype(CS) && istype(CH))
		if (CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
			return TRUE
	return FALSE
```

Thera-Pissed committed:
Update code/modules/mob/living/carbon/human/life.dm

Co-authored-by: Theos <theubernyan@gmail.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
short names are cleaner, also just found constants.fs

Thera-Pissed committed:
sprites!

Thera-Pissed committed:
filters actually filter most hazardous things, hcl for air alarms

before the power rework (I did) filters were kept short because long filter lists cost more power per filter. Now that it doesn't do that, there's no good reason not to filter most non air stuff.

Thera-Pissed committed:
comma

Thera-Pissed committed:
Update canister.dm

rye-rice commented:
Tell me if this still isn't finished, I'll merge in like 2 days after skimming the code if a review isn't given

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into chlorine-hydrogen-chloride-time

Thera-Pissed committed:
fixes sprite conflict

firebudgy commented:
Spilled a bottle of chlorine on a local and it runtimed.
![image](https://github.com/user-attachments/assets/5ba96ccd-b6a7-4ff9-a8d1-2d659d916abb)
![image](https://github.com/user-attachments/assets/77a41d83-69ef-44ee-8cad-d82f9b491ad2)

rye-rice commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -649,6 +652,33 @@
+	O.atmos_spawn_air("chlorine=[reac_volume/2];TEMP=[temp]")
+
+/datum/reagent/chlorine/expose_turf(turf/open/T, reac_volume)
+	if(istype(T))
```
Avoid single letter vars, changing obj/O to obj/exposed_object shouldn't mess anything up i think, same with turf/open/T

rye-rice commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -649,6 +652,33 @@
+	M.adjustOrganLoss(ORGAN_SLOT_STOMACH,1*REM)
+	. = 1
+	..()
+
```
Same here with M

Thera-Pissed committed:
chem review + runtime

changed chlorine to cl2, forgot to change it here. also addresses part of review

Thera-Pissed committed:
hubris

Thera-Pissed committed:
better var names part 2
